src/highlighting.js

```diff
@@ -65,12 +65,31 @@ function highlightResults(highlightFields, hit, pathToNested, include) {
   let mainHighlighted = false
   // Copy over all inline highlighted fields
   if (hit.highlight) {
-    _.each(highlightFields.inline, val => {
+    let { inline, inlineAliases } = highlightFields
+    // do the field replacement for the inline fields
+    _.each(inline, val => {
       if (hit.highlight[val]) {
         _.set(hit._source, val, hit.highlight[val][0])
         mainHighlighted = true
       }
     })
+    // do the field replacement for the inlineAliases fields
+    if (inlineAliases) {
+      for (let field in inlineAliases) {
+        let mapToField = inlineAliases[field]
+        // if we have a highlight result matching the inlineAliases TO field
+        if (hit.highlight[mapToField]) {
+          // if the field is only in inlineAliases OR it is in both but not inlined/highlighted already by the inline section
+          if (
```

Comment:
I'm not sure we need to bother with this check - if it's present on `inlineAlias`, we can safely assume it should be inlined and the cost of doing a `set` a second time is probably lower than the `includes` check anyway

Reply:
inlineAliases is really a backup rather than the norm. Since we do not know if the search is with `stemming` or not having `inline` take precedence over `inlineAliases` produces significantly better results than always overwrite. I tested this and this is why this is here.